cypress/integration/findings_spec.js

```diff
@@ -3,21 +3,221 @@
 
 describe('Findings', () => {
   before(() => {
-    // delete any existing indices
+    // Delete any existing indices
     cy.deleteAllIndices();
+
+    // Create test index
+    cy.createIndex('cypress-test-windows', TEST_INDEX);
+
+    // Create detector manually
+
+    // Visit Detectors page
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/detectors`);
+
+    cy.wait(10000);
+
+    // Locate Create detector button click to start
+    cy.contains('Create detector').click({ force: true });
+
+    // Check to ensure process started
+    cy.contains('Define detector');
+
+    // Enter a name for the detector in the appropriate input
+    cy.get(`input[placeholder="Enter a name for the detector."]`).type('test detector{enter}');
+
+    // Select our pre-seeded data source (cypress-test-windows)
+    cy.get(`[data-test-subj="define-detector-select-data-source"]`).type(
+      'cypress-test-windows{enter}'
+    );
+
+    // Select threat detector type (Windows logs)
+    cy.get(`input[id="windows"]`).click({ force: true });
+
+    // Wait for detector rules to load - timeout on click above ineffective
+    cy.wait(10000);
+
+    // Click Next button to continue
+    cy.get('button').contains('Next').click({ force: true }, { timeout: 2000 });
+
+    // Check that correct page now showing
+    cy.contains('Required field mappings');
+
+    // Select appropriate names to map fields to
+    for (let field_name in TEST_FIELD_MAPPINGS) {
+      const mappedTo = TEST_FIELD_MAPPINGS[field_name];
+
+      cy.contains('tr', field_name).within(() => {
+        cy.get(`[data-test-subj="detector-field-mappins-select"]`).select(mappedTo);
+      });
+    }
+
+    // Continue to next page
+    cy.get('button').contains('Next').click({ force: true }, { timeout: 2000 });
+
+    // Check that correct page now showing
+    cy.contains('Set up alerts');
+
+    // Type name of new trigger
+    cy.get(`input[placeholder="Enter a name for the alert condition."]`).type('test_trigger');
+
+    // Type rule name to trigger alert
+    cy.get(`[data-test-subj="alert-rulename-combo-box"]`).type('USB Device Plugged{enter}');
+
+    // Select applicable severity levels
+    cy.get(`[data-test-subj="security-levels-combo-box"]`).click({ force: true });
+    cy.contains('1 (Highest)').click({ force: true });
+
+    // Type rule severity to trigger alert
+    cy.get(`[data-test-subj="alert-severity-combo-box"]`).type('low{enter}');
+
+    // Continue to next page
+    cy.contains('Next').click({ force: true });
+
+    // Confirm page is reached
+    cy.contains('Review and create');
+
+    // Confirm field mappings registered
+    cy.contains('Field mapping');
+    for (let field in TEST_FIELD_MAPPINGS) {
+      const mappedTo = TEST_FIELD_MAPPINGS[field];
+
+      cy.contains(field);
+      cy.contains(mappedTo);
+    }
+
+    // Create the detector
+    cy.get('button').contains('Create').click({ force: true });
+
+    cy.wait(10000);
```

Comment:
Should we test for something after this wait?

Reply:
this is a pause to let the creation finish and the result to show, which is tested in the first 'it' block

Reply:
the creation of the detector is part of a big 'before()'

Reply:
ohh, okay

Reply:
can we export this into a method and call that inside the before block?

Reply:
done

Reply:
done